src/llama-context.h

```diff
@@ -27,7 +27,15 @@ struct llama_context {
 
     void synchronize();
 
-    const llama_model & get_model() const;
+    const llama_model   & get_model()   const;
+    const llama_cparams & get_cparams() const;
+
+    ggml_backend_sched_t get_sched() const;
+
+    ggml_context * get_ctx_compute() const;
+
+    // TODO: this method might be possible to avoid (seach for TAG_BACKENDS)
+    const std::vector<ggml_backend_ptr> & get_backends() const;
```

Comment:
I was wondering if we can avoid exposing the backends here. Currently, this is only needed in the `llama_kv_cache_unified::build_rope_shift()` method to determine the backend of the dequantized cache tensors:


https://github.com/ggml-org/llama.cpp/blob/ed7bb58a9eb811ccbcc7c659aa7ec5bbee9109e7/src/llama-kv-cache.cpp#L626-L637

@slaren Any ideas how to make this simpler?

Reply:
Thanks, fixed now.